web/pdf_attachment_viewer.js

```diff
@@ -49,152 +46,144 @@ var PDFAttachmentViewer = (function PDFAttachmentViewerClosure() {
+        viewerUrl = '?file=' + encodeURIComponent(blobUrl + '#' + filename);
+      } else if (PDFJSDev.test('CHROME')) {
+        // In the Chrome extension, the URL is rewritten using the history API
+        // in viewer.js, so an absolute URL must be generated.
+        // eslint-disable-next-line no-undef
+        viewerUrl = chrome.runtime.getURL('/content/web/viewer.html') +
+          '?file=' + encodeURIComponent(blobUrl + '#' + filename);
+      } else {
+        // Let Firefox's content handler catch the URL and display the PDF.
+        // In Firefox PDFJS.disableCreateObjectURL is always false, so
+        // blobUrl is always a blob:-URL and never a data:-URL.
+        viewerUrl = blobUrl + '?' + encodeURIComponent(filename);
       }
+      window.open(viewerUrl);
+      return false;
+    };
+  }
 
-      var names = Object.keys(attachments).sort(function(a, b) {
-        return a.toLowerCase().localeCompare(b.toLowerCase());
-      });
-      attachmentsCount = names.length;
-
-      for (var i = 0; i < attachmentsCount; i++) {
-        var item = attachments[names[i]];
-        var filename = removeNullCharacters(getFilenameFromUrl(item.filename));
-
-        var div = document.createElement('div');
-        div.className = 'attachmentsItem';
-        var button = document.createElement('button');
-        button.textContent = filename;
-        if (/\.pdf$/i.test(filename)) {
-          this._bindPdfLink(button, item.content, filename);
-        } else {
-          this._bindLink(button, item.content, filename);
-        }
+  /**
+   * @private
+   */
+  _bindLink(button, content, filename) {
+    button.onclick = (e) => {
+      this.downloadManager.downloadData(content, filename, '');
+      return false;
+    };
+  }
 
-        div.appendChild(button);
-        this.container.appendChild(div);
-      }
+  /**
+   * @param {PDFAttachmentViewerRenderParameters} params
+   */
+  render(params = Object.create(null)) {
+    var attachments = params.attachments || null;
+    var attachmentsCount = 0;
+
+    if (this.attachments) {
+      var keepRenderedCapability = params.keepRenderedCapability === true;
+      this.reset(keepRenderedCapability);
+    }
+    this.attachments = attachments;
 
+    if (!attachments) {
       this._dispatchEvent(attachmentsCount);
-    },
-
-    /**
-     * Used to append FileAttachment annotations to the sidebar.
-     * @private
-     */
-    _appendAttachment: function PDFAttachmentViewer_appendAttachment(item) {
-      this._renderedCapability.promise.then(function (id, filename, content) {
-        var attachments = this.attachments;
-
-        if (!attachments) {
-          attachments = Object.create(null);
-        } else {
-          for (var name in attachments) {
-            if (id === name) {
-              return; // Ignore the new attachment if it already exists.
-            }
+      return;
+    }
+
+    var names = Object.keys(attachments).sort((a, b) => {
```

Comment:
Nit: Unless I'm missing something, I don't think that you need an arrow function here (no `this` in the function scope).